gui-wm/wayfire/wayfire-0.4.0.ebuild

```diff
@@ -0,0 +1,97 @@
+# Copyright 2019-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit meson
+
+DESCRIPTION="A compiz like 3D wayland compositor"
+HOMEPAGE="https://github.com/WayfireWM/wayfire"
+
+if [[ ${PV} == 9999 ]]; then
+	inherit git-r3
+	EGIT_REPO_URI="https://github.com/WayfireWM/${PN}.git"
+	KEYWORDS=""
+else
+	SRC_URI="https://github.com/WayfireWM/${PN}/archive/${PV}.tar.gz -> ${P}.tar.gz"
+	KEYWORDS="~amd64 ~arm64 ~x86"
+fi
+
+LICENSE="MIT"
+SLOT="0"
+IUSE="+wcm +wf-shell +wf-config +wlroots +elogind systemd debug"
+
+DEPEND="
+		dev-libs/libevdev
+		dev-libs/libinput
+		media-libs/glm
+		media-libs/mesa:=[gles2,wayland,X]
+		media-libs/libjpeg-turbo
+		media-libs/libpng
+		media-libs/freetype:=[X]
+		x11-libs/libdrm
+		x11-libs/gtk+:3=[gtk,wayland,X]
+		x11-libs/cairo:=[X,svg]
+		x11-libs/libxkbcommon:=[X]
+		x11-libs/pixman
+		gui-libs/gtk-layer-shell
+		wf-config? ( ~gui-apps/wf-config-${PV} )
+		wlroots? ( >=gui-libs/wlroots-0.10.0[elogind=,systemd=,X] )
+"
+
+RDEPEND="
+		${DEPEND}
+		wcm? ( ~gui-apps/wcm-${PV} )
+		wf-shell? ( ~gui-apps/wf-shell-${PV} )
+		elogind? ( sys-auth/elogind )
+		systemd? ( sys-apps/systemd )
+		x11-misc/xkeyboard-config
+"
+
+BDEPEND="
+		${DEPEND}
+		virtual/pkgconfig
+		>=dev-libs/wayland-protocols-1.14
+"
+
+src_configure(){
+	local emsonargs=(
+		-Duse_system_wfconfig=$(usex wf-config true false)
+		-Duse_system_wlroots=$(usex wlroots true false)
+	)
+	if use debug; then
+		emesonargs+=(
+			"-Db_sanitize=address,undefined"
+		)
+	fi
+	meson_src_configure
+}
+
+pkg_preinst() {
+	if ! use systemd && ! use elogind; then
```

Comment:
You may want to do this in the init script? Not saying this is wrong though.

Reply:
I'm trying to keep this consistent with what sway does in their ebuild.
This way it is easier to manage as wayfire is being developed in parallel to sway changes.